### ISSUE TITLE
Fetch modules / binaries using psget, at runtime

### ISSUE THREAD

**Issue by ma499** [sponte/Powershell.Deployment#29]
Using PowerShell.Deployment in micro-services is incredibly distracting at code review time because it forms the majority of the files and lines of code in our service's repository. Moreover, we are now using PowerShell.Deployment with nodejs services, with no Visual Studio solution file, and so we can't easily use nuget to keep up-to-date.

Could PowerShell.Deployment be modified so that only a minimum of PowerShell scripts, relevant to the consuming project, need be committed to source control? One way to achieve this could be to use [psGet](http://psget.net/)

**Comment by sponte** [comment]
Hi @ma499, please feel free to submit pull request and we'll review

**Comment by slawomir-brzezinski-at-interxion** [comment]
Another solution for simpler, self-contained build environments (that don't depend on being able to download the scripts) could be to, rather than copy the files to the folder with sources on package-install, better to alter the projects build targets (nuget packages [can have their own .includes](https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#including-msbuild-props-and-targets-in-a-package)), to have the files copied from the `packages/` only during creation of package/artifact, and only to the respective staging folder for the packager.

**Comment by ma499** [comment]
Hi @sponte, what are your thoughts on general approach, though? Have been burnt previously working on a pull request that an upstream maintainer things does the wrong thing!

**Comment by sponte** [comment]
Hi @ma499, sorry for delay. I would be interested to understand how you're using these modules with node.js currently. From little information that I have, it seems like you're utilising powershell scripts to install/uninstall services on the windows server but don't want to use them in visual studio? 

Also, I have not been using this for a little while, @taliesins is a better person to comment as he's been actively developing and using these over the last 2 years or so.

**Comment by ma499** [comment]
Yes, we're using the powershell scripts to install/uninstall on a Windows server - specifically installing as a site in IIS and setting up monitoring via PRTG. We don't want to use a Visual Studio Solution just to nuget the PowerShell Deploy modules. So, for now, we did a copy/paste of the Deployment folder from a .NET project.
